axiom-eth/src/rlp/mod.rs

```diff
@@ -533,9 +687,34 @@ impl<'range, F: ScalarField> RlpChip<'range, F> {
                 len_len = self.gate().mul(ctx, len_len, field_in_list);
                 field_len = self.gate().mul(ctx, field_len, field_in_list);
             }
+
+
+
+            self.range.check_less_than_safe(ctx, field_len, (max_field_len + 1) as u64);
+
+
+            let field_cells = witness_subarray(
```

Comment:
Just a comment: this does not affect the circuit's soundness / completeness, since we always use the adjusted `field_len` for RLC constraint checks. But I agree it's nice to be consistent here.